### ISSUE TITLE
NIH Classification outputs are almost the same

### ISSUE THREAD

**Issue by 5aumit** [mlmed/torchxrayvision#177]
Hi!

I'm using the library's pretrained models for a project on the NIH Dataset. While inferring any image from the NIH dataset, I see probabilities (after sigmoid) almost the same. I'm not sure what's going wrong.

I'm just trying to use the NIH trained model to get outputs on the NIH **official** test set.

## Things I have tried
- With and without resizing to 224x224
- Using the same image loading from the `README.md`
- Using different weights apart from NIH

## Notes:
These are with `apply_sigmoid=False`, and I apply `torch,sigmoid()` after calling the model.

Labels : `array([0., 0., 1., 0., 0., 0., 0., 0., 0., 1., 0., 0., 0., 0.],`
Outputs: 
```
{'Atelectasis': np.float32(0.6440014),
 'Consolidation': np.float32(0.6361203),
 'Infiltration': np.float32(0.63221455),
 'Pneumothorax': np.float32(0.6273676),
 'Edema': np.float32(0.56724256),
 'Emphysema': np.float32(0.62366164),
 'Fibrosis': np.float32(0.62519747),
 'Effusion': np.float32(0.6296687),
 'Pneumonia': np.float32(0.6227707),
 'Pleural_Thickening': np.float32(0.63541263),
 'Cardiomegaly': np.float32(0.5024217),
 'Nodule': np.float32(0.62470883),
 'Mass': np.float32(0.6843057),
 'Hernia': np.float32(0.5326528),
 '': np.float32(0.62245935)}
```
## Context

I'm using an inherited `dataset` class because my NIH images are in subfolders, as that's how the official download was. I had to edit the `__getitem__()` method to account for the paths, but please let me know if I am passing an incorrectly processed image to the model.

```
class MyNIHDataset(xrv.datasets.NIH_Dataset):

  def __init__(self, imgpath, image_paths_df_path, *args, **kwargs):
    super().__init__(imgpath, *args, **kwargs)
    # Merge the provided image_paths_df with the dataset's csv
    # The base xrv.datasets.NIH_Dataset creates self.csv from Data_Entry_2017.csv
    # We merge on 'Image Index' from self.csv and 'image name' from image_paths_df
    image_paths_df = pd.read_csv(image_paths_df_path)
    image_paths_df['Absolute Path'] = image_paths_df.apply(lambda x : f"{NIH_path}/{x['parent path']}/{x['image name']}",axis=1)
    self.csv = self.csv.merge(image_paths_df[['image name', 'Absolute Path']],
                              left_on='Image Index', right_on='image name',
                              how='left')
    # Drop the redundant 'image name' column after merge if desired
    self.csv.drop(columns=['image name'], inplace=True)

  def __getitem__(self, idx):
        sample = {}
        sample["idx"] = idx
        sample["lab"] = self.labels[idx]

        img_path = self.csv.iloc[idx]['Absolute Path']
        # img = skimage.io.imread(img_path)
        img = xrv.utils.load_image(img_path)

        sample["img"] = img


        if self.pathology_masks:
            # Ensure imgid is correctly extracted for mask loading if needed
            imgid = self.csv['Image Index'].iloc[idx]
            sample["pathology_masks"] = self.get_mask_dict(imgid, sample["img"].shape[2])

        sample = xrv.datasets.apply_transforms(sample, self.transform)
        sample = xrv.datasets.apply_transforms(sample, self.data_aug)

        return sample

# Instantiate NIHDataset with the image_paths_df
# transform = torchvision.transforms.Compose([xrv.datasets.XRayCenterCrop(),xrv.datasets.XRayResizer(224)])
transform = torchvision.transforms.Compose([xrv.datasets.XRayCenterCrop()])
nih_df = MyNIHDataset(imgpath=NIH_path, views=["PA", "AP"], unique_patients=False, image_paths_df_path=image_paths_csv,transform = transform)
```

For ease of use, this is how I'm loading my model
```
def load_model(dataset_name = 'NIH',apply_sigmoid = False):
  """
  Loads a pre-trained DenseNet model from torchxrayvision based on the specified dataset.

  Args:
    dataset_name (str, optional): The name of the dataset for which to load the model.
                                   Defaults to 'NIH'. Supported datasets include:
                                   'RSNA', 'NIH', 'PadChest', 'CheX', 'MIMIC_NB', 'MIMIC_CH'.

  Returns:
    torchvision.models.densenet.DenseNet: A pre-trained DenseNet model.
  """
  ds_dict = {
      'All' : "densenet121-res224-all", # All
      'RSNA' : "densenet121-res224-rsna", # RSNA Pneumonia Challenge
      'NIH' : "densenet121-res224-nih", # NIH Chest X-Ray8
      'PadChest' : "densenet121-res224-pc", # PadChest
      'CheX' : "densenet121-res224-chex", # CheXpert
      'MIMIC_NB' : "densenet121-res224-mimic_nb", # MIMIC-CXR
      'MIMIC_CH' : "densenet121-res224-mimic_ch"
  }

  # Load the DenseNet model with specified weights
  model = xrv.models.DenseNet(weights=ds_dict[dataset_name],apply_sigmoid=apply_sigmoid)

  # Move the model to the globally defined device (GPU if available, else CPU)
  model.to(device)

  return model
```

Finally running inference:
```
model = load_model('All',False)#.cuda()
img = nih_df[1000]['img']
img = torch.from_numpy(img)
out = infer_model(img,model)
```

**Comment by ieee8023** [comment]
Maybe use the load images method from this example? https://github.com/mlmed/torchxrayvision/blob/main/scripts/process_image.py

You can compare your pre-processing with the outputs for these test files: https://github.com/mlmed/torchxrayvision/tree/main/scripts

**Comment by 5aumit** [comment]
Thanks for your quick response.

I did update my `dataset` class. The `__getitem__` is below. I'm using the `xrv.utils.load_image` method.

```
def __getitem__(self, idx):
        row = self.dataframe.iloc[idx]
        img_path = row[self.image_path_col]
        image_name = row['Image Index']

        # Load image using xrv.utils.load_image as it handles different image types
        img = xrv.utils.load_image(img_path)

        # Apply transformations. self.transform should now include ToTensor().
        img_tensor = self.transform(img)

        # Extract ground truth labels for the specified findings
        ground_truth = torch.tensor([row[finding] for finding in self.findings_list], dtype=torch.float32)

        # Return as a tuple: (image_tensor, ground_truth_tensor, image_name_string, image_path_string)
        return img_tensor, ground_truth, image_name, img_path
```
Object (the only transform is cropping):
```
standalone_transform = torchvision.transforms.Compose([
        xrv.datasets.XRayCenterCrop(),
    ])
inference_dataset = NIHInferenceDataset(test_df, 'Absolute Path', findings_list, transform=standalone_transform)
```

I tried inferring an image which has Atelectasis and Consolidation in the NIH labels.
```
img = inference_dataset[24][0]
model = xrv.models.get_model("densenet121-res224-nih")
with torch.no_grad():
    img = torch.from_numpy(img).unsqueeze(0)
    preds = model(img).cpu()
    output["preds"] = dict(
        zip(xrv.datasets.default_pathologies, preds[0].detach().numpy())
    )
display(output)
```
But the outputs have almost every category ~0.5
```
 Warning: Input size (1024x1024) is not the native resolution (224x224) for this model. A resize will be performed but this could impact performance.
{'preds': {'Atelectasis': np.float32(0.63393676),
  'Consolidation': np.float32(0.5295638),
  'Infiltration': np.float32(0.550092),
  'Pneumothorax': np.float32(0.5066911),
  'Edema': np.float32(0.5003616),
  'Emphysema': np.float32(0.50357413),
  'Fibrosis': np.float32(0.52621615),
  'Effusion': np.float32(0.8831826),
  'Pneumonia': np.float32(0.5077727),
  'Pleural_Thickening': np.float32(0.53475565),
  'Cardiomegaly': np.float32(0.46323994),
  'Nodule': np.float32(0.5273855),
  'Mass': np.float32(0.52555776),
  'Hernia': np.float32(0.5171931),
  'Lung Lesion': np.float32(0.5),
  'Fracture': np.float32(0.5),
  'Lung Opacity': np.float32(0.5),
  'Enlarged Cardiomediastinum': np.float32(0.5)}}
```

Am I missing something?
Note - I understand these might be logits since `apply_sigmoid` is set to `False` by default in the `xrv.models.get_model` method. But even then, should the logits be so similar for every single class?

**Comment by ieee8023** [comment]
Which test image are you running it on? Is that close to the expected output?
